components/map/basic/src/index.js

```diff
@@ -0,0 +1,227 @@
+    this._renderHereMap()
+  }
+
+  _renderHereMap () {
+    const map = this._buildMapInstance()
+    if (map) {
+      this.props.isInteractive && this._makeInteractive()
+      this.props.isInteractive && this._buildMapUIComponents()
+      this.props.marker && this._buildMarkers()
+      this.props.circleShape && this._buildCircleShape()
+    }
+  }
+
+  _buildMarkers () {
+    const generalIcon = new window.H.map.DomIcon(this.props.marker.genericIcon)
+    this.props.marker.elements.forEach(element => {
+      const icon = element.icon ? new window.H.map.DomIcon(element.icon) : generalIcon
+      this.map.instance.addObject(new window.H.map.DomMarker({ lat: element.lat, lng: element.lng }, { icon: icon }))
+    })
+  }
+
+  _buildCircleShape () {
+    this.props.circleShape.elements.forEach(element => {
+      this.map.instance.addObject(
+        new window.H.map.Circle({lat: element.lat, lng: element.lng},
+        element.radius,
+        { style: element.style || this.props.circleShape.styleGeneric })
+      )
+    })
+  }
+
+  _buildMapInstance () {
+    const DOMElementToBeFilled = document.getElementById('sui-here-map')
+    if (DOMElementToBeFilled) {
+      this.map.instance = new window.H.Map(
+        DOMElementToBeFilled,
+        this.map.defaultLayers.normal.map,
+        {...this.props.options}
+      )
+    }
+
+    return this.map.instance
+  }
+
+  componentDidMount () {
+    if (window.H && !this.map.instance) {
+      this._renderHereMap()
+    }
+  }
+
+  render () {
+    return (
+      <div id='sui-here-map' className='sui-MapBasic' />
+    )
+  }
+}
+
+MapBasic.displayName = 'MapBasic'
+
+MapBasic.propTypes = {
+  /**
+   * An array with the link of the HERE libraries that we would like to load. By default is gonna load core, service, ui and mapevents. Useful if there are cdn or version changes
+   */
+  libraries: PropTypes.array,
+  /**
+   * An array of the here styles that we desire to load.
+   */
+  styles: PropTypes.array,
+  /**
+   * An object with the definition of our map. All options of here maps are allowed, for example
+   * { zoom: <number>, center: { lat: number, lng: number} }
+   */
+  options: PropTypes.object.isRequired,
+  /**
+   * Our appid token, provided by HERE maps
+   */
+  appId: PropTypes.string.isRequired,
+  /**
+   * Our appCode token, provided by HERE maps
+   */
+  appCode: PropTypes.string.isRequired,
+  /**
+   * If is a interactable map (drag, zoom)... With this enabled the UI will be automatically loaded
+   */
+  isInteractive: PropTypes.bool,
+  /**
+   * Use CIT ONLY SHOULD BE ENABLED if we are going to use a customer integration testing app id and app code
+   * More info about CIT here https://developer.here.com/documentation/venue-maps/topics/request-cit-environment.html
+   */
+  useCIT: PropTypes.bool,
+  /**
+   * Enables or disables HTTPS communication with the platform.
+   */
+  useHTTPS: PropTypes.bool,
```

Comment:
I really this it's reponsiblity of the component to determine if it's in an https context or not.
Don't you think?